File: backend/src/belfort/symbols.py
"""Symbol normalization for API and CLI."""

from __future__ import annotations


def normalize_symbol(symbol: str) -> str:
    s = symbol.strip().upper().replace("/", "").replace("-", "")
    if s.endswith(("USDT", "USD", "BUSD")):
        return s
    return f"{s}USDT"


def display_symbol(symbol: str) -> str:
    s = symbol.strip().upper()
    for suffix in ("USDT", "BUSD", "USD"):
        if s.endswith(suffix) and len(s) > len(suffix):
            return s[: -len(suffix)]
    return s

File: frontend/src/lib/symbols.ts
/** Map UI symbol (BTC) to exchange pair (BTCUSDT) for the API. */
export function toApiSymbol(symbol: string): string {
  const s = symbol.trim().toUpperCase().replace(/[^A-Z0-9]/g, "");
  if (s.endsWith("USDT") || s.endsWith("USD") || s.endsWith("BUSD")) return s;
  return `${s}USDT`;
}

/** Display label without quote suffix. */
export function toDisplaySymbol(symbol: string): string {
  const s = symbol.trim().toUpperCase();
  return s.replace(/USDT$|USD$|BUSD$/, "") || s;
}
